### ISSUE TITLE
Recommendation for using enums instead of checking for strings !

### ISSUE THREAD

**Issue by yuehernkang** [nus-cs2030/2122-s1#275]
## Description 
Describe the problem and code snippets if necessary 

I saw there is a thread asking about using 'instanceof' in the projects and the reply was we could check the strings to find out the different events (ArrivalEvent, ServeEvent, etc.)

Let me suggest using Enums instead! This way, we wont have to check for any spelling mistakes anywhere in our code and we will not have any runtime errors.

This makes it easier to debug as the compiler will check for us to make sure the spelling is correct ! 

Here is an example from my project for the state of the Server:

`public enum ServerState {
    IDLE,
    SERVING
}
`

**Comment by Hongyi6328** [comment]
Nice way! thank you very much.
May I know how you associate event with event types? Is it something like this:
```
event.getEventType() == EventType.ArriveEvent;
```

**Comment by yuehernkang** [comment]
> Nice way! thank you very much. May I know how you associate event with event types? Is it something like this:
> 
> ```
> event.getEventType() == EventType.ArriveEvent;
> ```

yes. But I believe usually the convention for enums are capital letters. 
Therefore it can look something like this:
`event.getEventType() == EventType.ARRIVE;`
